### PR TITLE
WIP: Quantiles of vector of infinities

### PR DESCRIPTION
**Pull request by alexhallam**
A fix for issue #19542

### CODE REVIEW

File: base/statistics.jl [hunk 1]
```diff
@@ -683,7 +683,11 @@ end
     else
         a = T(v[i])
         b = T(v[i+1])
-        return a + h*(b-a)
+	if isfinite(a) && isfinite(b)
```
Comment (tkelman) [inline base/statistics.jl]: spaces, not tabs

**Comment by andreasnoack** [comment]
Looks goog but needs a rebase.

**Comment by alexhallam** [comment]
Sorry, @andreasnoack, this is embarrassing. Do you have a reference on how to `rebase` this? Thanks for the patience.

**Comment by andreasnoack** [comment]
Try searching for "github rebase PR" or similar. Most of the hits might give more information than necessary. The steps I do are

1. `git checkout [PR branch]`
2. `git fetch origin`
3. `git rebase origin/master`
4. ...CONFLICTS...
5. Search for places with `<<<` in `base/statistics.jl` and `test/statistics.jl` and resolve the conflicts
6. `git add base/statistics.jl test/statistics.jl`
7. `git rebase --continue`
8. `git push --force fork [PR branch]`

In theory, I should also be able to do this here on GitHub but the web editor hangs.

File: base/statistics.jl [hunk 2]
```diff
@@ -675,15 +675,18 @@ end
...
-
     i = trunc(Int,t0) + 1
 
     if h == 0
         return T(v[i])
     else
         a = T(v[i])
         b = T(v[i+1])
-        return a + ifelse(a == b, zero(a), h*(b-a))
+	if isfinite(a) && isfinite(b)
```
Comment (tkelman) [inline base/statistics.jl]: agains, please use (multiples of) 4 space indent, not tabs

**Review by tkelman — DISMISSED** [review]

**Comment by certifiedloud** [comment]
:tada:

**Pull request merged**